src/pages/patterns/text-toolbar-pattern/index.mdx

```diff
@@ -0,0 +1,276 @@
+
+</Column>
+</Row>
+
+1. **Actions:** Use “Undo” and “Redo” to undo or revert the last change made. Use “Cut”, “Copy”, and “Paste” to move pieces of text to another place within the text area, to copy text to the clipboard, or to paste copied text from the clipboard to a different place within the text area.
+2. **Formatting:** Change the typeface, size, style, and color of text.
+3. **Paragraph:** Select different alignments and indents for paragraph text and indicate bulleted, checked, or numbered lists.
+4. **Attachment:** Attach files or embed links in strings of text.
+5. **Search:** Search key words within existing paragraphs of text.
+6. **Text area:** Designated area to type editable text.
+
+## When to use
+
+The text toolbar provides an efficient way for users to perform several tasks:
+
+- Create, edit and save simple text files quickly.
+- Edit text with common actions like cut, copy and paste and formatting text styles.
+- Perform basic text search functionality.
+- Attach files or embed links within text.
+
+## Behaviors
+
+### States
+
+The text toolbar includes a series of basic actions that adopt the [icon button](https://www.carbondesignsystem.com/components/button/usage) style. The action buttons typically have five different states—enabled, hover, focus, active, and disabled.
+
+<Row>
+<Column colLg={12}>
+
+![Text toolbar button states.](/images/text-toolbar-button-states.png)
+
+</Column>
+</Row>
+
+The typeface and type size menus adopt the [dropdown](https://www.carbondesignsystem.com/components/dropdown/usage/#dropdown) style. The user can easily choose any typeface and type size to customize their text.
+
+<Row>
+<Column colLg={8}>
+
+![Text toolbar dropdowns.](/images/text-toolbar-dropdowns.png)
+
+</Column>
+</Row>
+
+The text color and paragraph alignment controls open a menu that displays different options to choose from.
+
+<Row>
+<Column colLg={8}>
+
+![Text toolbar menus.](/images/text-toolbar-menus.png)
+
+</Column>
+</Row>
+
+### Attach files
+
+The user can upload and attach a file from their computer. Once a file has been chosen, the file will load in the bottom left of the text area. A file can be removed by clicking the close `x` icon.
+
+<Row>
+<Column colLg={8}>
+
+![Attaching files to the text area.](/images/text-toolbar-attachment.png)
+
+</Column>
+</Row>
+
+### Embed links
+
+Links can be embedded within strings of text. To embed a link, click on the link button to open the text field.
+
+<Row>
+<Column colLg={8}>
+
+![Embedding text links.](/images/text-toolbar-link-open.png)
+
+</Column>
+</Row>
+
+Enter the html link in the text field and then click the link button to embed the link. To exit the text field, click the close `x` icon or anywhere outside of the field. To reopen the link text field to edit or delete, highlight the inline link text with the cursor.
+
+<Row>
+<Column colLg={8}>
+
+![Embedding text links.](/images/text-toolbar-link-embed.png)
+
+</Column>
+</Row>
+
+### Search
```

Comment:
This feature seems a little tricky to do accessibly, I'm not sure how someone would be able to use the keyboard to interact with the results in the WYSIWYG. It seems like from the linked WAI ARIA example that the editor would need to be a tab stop and would not have tab stops inside of it since it is a `textarea`

Reply:
That is a good to note. 👍🏻 When someone eventually starts to build this in code (whether it is us, or a contribution from the community) we can expand/edit this depending on the  capabilities. This seemed to be a use case that has been needed which is why we are including it for now.

Reply:
@laurenmrice the main concern on my end is that we run into a situation where functionality is expected but is difficult or impossible to be made accessible. We ran into this with our own work this past year and it's important that our process reflects this so that we don't run into this problem again, either with our team or teams building this.

In-industry examples often don't reflect that something can be built accessibly. I think Wordpress' work on their new editor is a great example of this and how process or culture results in a built experience that is difficult or impossible to use for an entire set of users.

Reply:
@joshblack thank you so much for mentioning Wordpress' big gnarly PR disaster of a non-accessible WYSIWYG 👏 👏 👏 

Did a few minutes of preliminary research on some of the go-to WYSIWYG libraries (in particular [TinyMCE](https://github.com/tinymce/tinymce)) available in JS land and an in context search functionality was notable by it's absence.

Reply:
Thanks for the chat, @laurenmrice! Helped a ton, will post any research I find too down below based on what @dakahn shared 👀